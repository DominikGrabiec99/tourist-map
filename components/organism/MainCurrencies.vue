<template>
  <div class="currency-box flex flex-row">
    <template v-for="currency in sortedCurrencies" :key="currency.code">
      <MoleculeCurrencyInformation v-if="currency.mid" :currency="currency" />
    </template>
  </div>
</template>

<script setup lang="ts">
/** INTERFACES */
import IMainCurrency from '@/ts/interfaces/MainCurrency';

const { currencies = [] } = defineProps<{
  currencies : IMainCurrency[];
}>();

const sortedCurrencies = computed(() =>
  currencies.sort((currencyA, currencyB) =>
    currencyA.priority > currencyB.priority ? 1 : -1
  )
);
</script>

<style scoped lang="scss">
.currency-box {
  position: relative;
  animation: currencyRole 40s 0.5s linear infinite;
}
@keyframes currencyRole {
  0% {
    transform: translateX(0%);
  }

  100% {
    transform: translateX(-100%);
  }
}
</style>
