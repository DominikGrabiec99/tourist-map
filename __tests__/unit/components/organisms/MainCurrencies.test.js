import { describe, it } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import MainCurrencies from '@/components/organism/MainCurrencies.vue';

import { mainCurrencies } from '@/__tests__/mockData';

const sortedMainCurrencies = mainCurrencies.sort((currencyA, currencyB) =>
  currencyA.priority > currencyB.priority ? 1 : -1
);

describe('MainCurrencies.vue', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallowMount(MainCurrencies, {
      props: {
        currencies: mainCurrencies,
      },
    });
  });

  it('render component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('check sorted currencies', () => {
    expect(wrapper.vm.sortedCurrencies).toEqual(sortedMainCurrencies);
  });
});
