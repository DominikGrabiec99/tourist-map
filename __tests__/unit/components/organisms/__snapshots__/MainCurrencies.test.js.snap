// Vitest Snapshot v1

exports[`MainCurrencies.vue > render component 1`] = `
<div
  class="currency-box flex flex-row"
  data-v-a6221edb=""
>
  
  
  <moleculecurrencyinformation
    currency="[object Object]"
    data-v-a6221edb=""
  />
  
  
  <moleculecurrencyinformation
    currency="[object Object]"
    data-v-a6221edb=""
  />
  
  
</div>
`;
